app/app/src/main/java/com/example/myapplication/MainActivity.java

```diff
@@ -468,4 +483,25 @@ public void run() {
         Thread animationInteractionThread = new Thread(animationInteraction);
         animationInteractionThread.start();
     }
-}
\ No newline at end of file
+
+    public void connectionHelper(final String name) {
+        Runnable animationInteraction = new Runnable() {
```

Comment:
It would be good to name the Runnable something more relevant to the situation, for readability :) `animationInteraction` makes it sound like this Runnable relates to the animation parts of the app.